### PR TITLE
feat: improvements

### PR DESCRIPTION
**Pull request by nabeelsaabna**
- feat: replace `RUNINSTALL_MATCH` with `RUNINSTALL_INCLUDES` 
- feat: add `RUNINSTALL_EXCLUDES`
- feat: add `RUNINSTALL_ENABLE_NO_GIT`
- chore: add tests
- fix: graceful error handling
- fix: add default tools to Dockerfile

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -82,4 +84,8 @@ function delegateCommand() {
   });
 
   shutdown(res.status);
+}
+
+(async function () {
+  await main();
 })();
```
Comment (viceice) [inline src/index.js:91]: why this change? I don't see any benefit

**Review by viceice — COMMENTED** [review]
this PR is very big, should be split into smaller parts for easier review

File: README.md [hunk 2]
```diff
@@ -4,16 +4,22 @@
...
+ 
 Example:
 
 ```
-export RUNINSTALL_MATCH=org_a,org_b
+export RUNINSTALL_INCLUDES=org_a,org_b
 mvn --version
 ```
 
+`RUNINSTALL_MATCH` is deprecated and replaced by `RUNINSTALL_INCLUDES`.
```
Comment (PhilipAbed) [inline README.md:21]:
```suggestion
`RUNINSTALL_MATCH` is deprecated and has been replaced by `RUNINSTALL_INCLUDES`.
```

File: src/index.js (hunk 1 above)
Comment (PhilipAbed) [reply src/index.js:91]: it's more readable i believe, we would like to see "main" so we know where the code is being executed.

doesn't hurt

**Review by PhilipAbed — APPROVED** [review]

File: src/index.js (hunk 1 above)
Comment (nabeelsaabna) [reply src/index.js:91]: yes, just readability 
Javascript is not easy to read

**Comment by nabeelsaabna** [comment]
@viceice 
sorry, PR size is indeed not small.. its due to time limitation

File: README.md [hunk 3]
```diff
@@ -4,16 +4,22 @@
 
 ### Path matching
 
-If `RUNINSTALL_MATCH` is undefined, or does not match, then all logic will be skipped.
-The command will be passed through to `/usr/local/bin/${cmd}`.
+- `RUNINSTALL_ENABLE_NO_GIT`: if set to `true` it allows running tool update without any git repository. it will bypass `RUNINSTALL_INCLUDES` and `RUNINSTALL_EXCLUDES`
```
Comment (PhilipAbed) [inline README.md:7]:
```suggestion
- `RUNINSTALL_ENABLE_NO_GIT`: if set to `true`, it allows running tool update without any git repository. it will bypass `RUNINSTALL_INCLUDES` and `RUNINSTALL_EXCLUDES`
```

Comment (PhilipAbed) [reply README.md:7]: me being Justo

**Review by rarkins — APPROVED** [review]
